src/gui/tray/activitylistmodel.cpp

```diff
@@ -343,6 +347,48 @@ void ActivityListModel::activitiesReceived(const QJsonDocument &json, int status
         a._dateTime = QDateTime::fromString(json.value("datetime").toString(), Qt::ISODate);
         a._icon = json.value("icon").toString();
 
+        auto richSubjectData = json.value("subject_rich").toArray();
+        a._subjectRich = richSubjectData[0].toString();
+        auto parameters = richSubjectData[1].toObject();
+        const QRegularExpression subjectRichParameterRe("({[a-zA-Z0-9]*})");
+        const QRegularExpression subjectRichParameterBracesRe("[{}]");
+
+        for (auto i = parameters.begin(); i != parameters.end(); i++) {
+            const auto parameterJsonObject = i.value().toObject();
+            Activity::RichSubjectParameter parameter = {
+                parameterJsonObject.value("type").toString(),
+                parameterJsonObject.value("id").toString(),
+                parameterJsonObject.value("name").toString(),
+                QString(),
+                QUrl(),
+            };
+
+            if(parameter.type == "file") {
+                parameter.path = parameterJsonObject.value("path").toString();
+            }
+
+            if(parameter.type == "file" && parameterJsonObject.contains("link")) {
+                parameter.link = QUrl(parameterJsonObject.value("link").toString());
+            }
+
+            a._subjectRichParameters[i.key()] = parameter;
+        }
+
+        if(!a._subjectRich.isEmpty()) {
+            QRegularExpressionMatchIterator i = subjectRichParameterRe.globalMatch(a._subjectRich);
+
+            while (i.hasNext()) {
+                const QRegularExpressionMatch match = i.next();
+                QString word = match.captured(1);
```

Comment:
@claucambra `QString word` -> `auto word`

Reply:
Thanks, changed to autio